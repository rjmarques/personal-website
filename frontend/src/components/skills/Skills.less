@import "../../theme.less";

.skills {
  .skills-card-row {
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;

    grid-template-columns: repeat(2, 1fr);

    .for-size(laptop-up; {
      grid-template-columns: repeat(3, 1fr);
    });

    a {
      text-decoration: none;
      border-color: @light-grey;

      &:focus {
        border-color: @primary-color;
      }
    }

    .skills-card {
      border-color: inherit;
      font-family: inherit;
      padding: 20px 20px 0px 20px;

      img {
        max-height: 80px;
      }
    }

    .skills-card-meta {
      font-weight: 700;
      letter-spacing: 2px;
      text-align: center;
      text-transform: uppercase;

      .for-size(tablet-up; {
        font-size: 18px;
      });
    }
  }
}
