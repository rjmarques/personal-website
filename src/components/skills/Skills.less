@import "../../theme.less";

@verticalGutter: 16px;

.Skills-card-row {
  margin-bottom: -@verticalGutter;

  a {
    text-decoration: none;
  }
}

.Skills-card {
  border-color: @light-grey;
  font-family: inherit;
  margin-bottom: @verticalGutter;
  padding: 20px 20px 0px 20px;

  img {
    max-height: 80px;
  }
}

.Skills-card-meta {
  font-weight: 700;
  font-size: 18px;
  letter-spacing: 2px;
  text-align: center;
  text-transform: uppercase;
}
