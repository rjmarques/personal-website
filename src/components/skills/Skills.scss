@use "../../theme.scss" as *;

.skills {
  .skills-card-row {
    display: grid;
    grid-column-gap: 16px;
    grid-row-gap: 16px;
    grid-template-columns: repeat(2, 1fr);

    @include for-size(laptop-up) {
      grid-template-columns: repeat(3, 1fr);
    }

    .skill-link {
      background-color: $white;
      color: inherit;
      border: 1px solid $light-grey;
      text-decoration: none;
      transition: all 0.3s ease;

      &:focus {
        border-color: $primary-color;
        outline: none;
      }

      &:hover {
        box-shadow: 0 5px 38px 2px rgba(0, 0, 0, 0.1);
      }
    }

    .skills-card {
      border: none;
      box-shadow: none;
      background: transparent;
      padding: 20px;
      height: 100%;
      display: flex;
      flex-direction: column;

      .skill-image {
        max-height: 80px;
        object-fit: contain;
      }

      .skills-card-meta {
        font-weight: 700;
        letter-spacing: 2px;
        margin-top: 24px;
        margin-bottom: 5px;
        padding: 0;
        text-align: center;
        text-transform: uppercase;

        @include for-size(tablet-up) {
          font-size: 18px;
        }
      }
    }
  }
}

