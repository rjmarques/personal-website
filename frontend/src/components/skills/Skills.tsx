import { Card } from "antd";
import React from "react";

import ContentSection from "../common/ContentSection";

import "./Skills.less";

interface ISkill {
  icon: any;
  title: string;
  url: string;
}

const skills: ISkill[] = [
  {
    icon: require("./img/react.svg"),
    title: "react",
    url: "https://reactjs.org/",
  },

  {
    icon: require("./img/typescript.svg"),
    title: "typescript",
    url: "https://www.typescriptlang.org/",
  },
  {
    icon: require("./img/go.svg"),
    title: "golang",
    url: "https://golang.org/",
  },
  {
    icon: require("./img/postgres.svg"),
    title: "postgresql",
    url: "https://www.postgresql.org/",
  },
  {
    icon: require("./img/java.svg"),
    title: "java ee",
    url: "https://www.oracle.com/technetwork/java/javaee/overview/index.html",
  },
  {
    icon: require("./img/javascript.svg"),
    title: "javascript",
    url: "https://www.javascript.com/",
  },
  {
    icon: require("./img/angular.svg"),
    title: "angular",
    url: "https://angular.io/",
  },
  {
    icon: require("./img/docker.svg"),
    title: "docker",
    url: "https://www.docker.com/",
  },
];

const Skills = () => (
  <ContentSection
    title="Expertise"
    subTitle="The technologies I am proficient in"
    wrapperClass="skills"
  >
    <div className="skills-card-row materialize">
      {skills.map((skill: ISkill) => (
        <a
          key={skill.title}
          href={skill.url}
          target="_blank"
          rel="noopener noreferrer"
        >
          <Card
            className="skills-card"
            hoverable={true}
            cover={<img alt={skill.title} src={skill.icon} />}
          >
            <div className="skills-card-meta">{skill.title}</div>
          </Card>
        </a>
      ))}
    </div>
  </ContentSection>
);

export default Skills;
