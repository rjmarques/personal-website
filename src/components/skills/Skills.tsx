import { Card, Col, Row } from "antd";
import React from "react";

import ContentSection from "../common/ContentSection";

import "./Skills.less";

interface ISkill {
  icon: any;
  title: string;
  url: string;
}

const skills: ISkill[] = [
  {
    icon: require("./img/angular.svg"),
    title: "angular",
    url: "https://angular.io/"
  },
  {
    icon: require("./img/react.svg"),
    title: "react",
    url: "https://reactjs.org/"
  },
  {
    icon: require("./img/javascript.svg"),
    title: "javascript",
    url: "https://www.javascript.com/"
  },
  {
    icon: require("./img/sass.svg"),
    title: "sass",
    url: "https://sass-lang.com/"
  },
  {
    icon: require("./img/typescript.svg"),
    title: "typescript",
    url: "https://www.typescriptlang.org/"
  },
  {
    icon: require("./img/java.svg"),
    title: "java ee",
    url: "https://www.oracle.com/technetwork/java/javaee/overview/index.html"
  },
  {
    icon: require("./img/postgres.svg"),
    title: "postgresql",
    url: "https://www.postgresql.org/"
  },
  {
    icon: require("./img/c++.svg"),
    title: "c++",
    url: "http://www.cplusplus.com/"
  }
];

const Skills = () => (
  <ContentSection
    title="Expertise"
    subTitle="The technologies I am proficient in"
    wrapperClass="Skills"
  >
    <Row gutter={16} className="Skills-card-row materialize">
      {skills.map((skill: ISkill) => (
        <Col key={skill.title} xs={24} sm={12} md={12} lg={6} xl={6}>
          <a href={skill.url} target="_blank">
            <Card
              className="Skills-card"
              hoverable={true}
              cover={<img alt={skill.title} src={skill.icon} />}
            >
              <div className="Skills-card-meta">{skill.title}</div>
            </Card>
          </a>
        </Col>
      ))}
    </Row>
  </ContentSection>
);

export default Skills;
