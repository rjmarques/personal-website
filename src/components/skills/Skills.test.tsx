import React from "react";
import renderer from "react-test-renderer";

import Skills from "./Skills";

describe("renders without crashing", () => {
  it("renders correctly", async () => {
    const component = renderer.create(<Skills />);

    const tree = component.toJSON();
    expect(component).toMatchSnapshot();
  });
});
