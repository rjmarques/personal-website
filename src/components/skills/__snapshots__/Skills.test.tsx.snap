// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing renders correctly 1`] = `
<section
  className="Skills"
>
  <div
    className="content"
  >
    <div
      className="Content-title"
    >
      <h2>
        Expertise
      </h2>
      <span
        className="Section-sub-title"
      >
        The technologies I am proficient in
      </span>
    </div>
    <div
      className="ant-row Skills-card-row materialize"
      style={
        Object {
          "marginLeft": -8,
          "marginRight": -8,
        }
      }
    >
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://reactjs.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="react"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "react.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                react
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://angular.io/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="angular"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "angular.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                angular
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://www.javascript.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="javascript"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "javascript.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                javascript
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://sass-lang.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="sass"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "sass.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                sass
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://www.typescriptlang.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="typescript"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "typescript.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                typescript
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://www.oracle.com/technetwork/java/javaee/overview/index.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="java ee"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "java.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                java ee
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://golang.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="golang"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "go.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                golang
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="ant-col-xs-24 ant-col-sm-12 ant-col-md-12 ant-col-lg-6 ant-col-xl-6"
        style={
          Object {
            "paddingLeft": 8,
            "paddingRight": 8,
          }
        }
      >
        <a
          href="https://www.postgresql.org/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <div
            className="ant-card Skills-card ant-card-bordered ant-card-hoverable"
          >
            <div
              className="ant-card-cover"
            >
              <img
                alt="postgresql"
                src={
                  Object {
                    "ReactComponent": Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "render": [Function],
                    },
                    "__esModule": true,
                    "default": "postgres.svg",
                  }
                }
              />
            </div>
            <div
              className="ant-card-body"
              style={Object {}}
            >
              <div
                className="Skills-card-meta"
              >
                postgresql
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</section>
`;
